src/js/jsx/search/SearchBar.jsx

```diff
@@ -213,82 +216,131 @@ define(function (require, exports, module) {
+                    // show that filter option
+                    if (option.id.indexOf("FILTER") === 0 && _.isEqual(this.state.filter, category)) {
+                        return false;
+                    }
+                    
+                    option.priority++;
+
+                    if (this.state.filter.length > 0) {
+                        // All terms in this.state.filter must be in the option's category
+                        _.forEach(this.state.filter, function (filterValue) {
+                            if (!_.contains(category, filterValue)) {
+                                useTerm = false;
+                            }
+                        });
+
+                        if (!useTerm) {
+                            return false;
+                        }
+                    }
+
+                    // If haven't typed anything, want to use everything that fits into the category
+                    if (searchTerm === "") {
+                        count++;
+                        return true;
+                    }
 
-                if (this.state.filter.length > 0) {
-                    // All terms in this.state.filter must be in the option's category
-                    _.forEach(this.state.filter, function (filterValue) {
-                        if (!_.contains(category, filterValue)) {
-                            useTerm = false;
+                    option.priority++;
+
+                    // If option has a path, search for it with and without '/' characters
+                    // This first check preserves order. Later we check for every word,
+                    // but it will have a lower priority than if it matches perfectly
+                    var pathInfo = option.pathInfo ? option.pathInfo.toLowerCase() : "",
+                        searchablePath = pathInfo.concat(pathInfo.replace(/\//g, " "));
+                    
+                    if (searchablePath.indexOf(searchTerm) > -1) {
+                        // If option is the autofill option, should jump to the top
+                        if (option.id === autofillID) {
+                            option.priority = 0;
+                        }
+                        return true;
+                    }
+
+                    option.priority++;
+                    
+                    var searchTerms = searchTerm.split(" "),
+                        numTermsInTitle = 0;
+                    useTerm = false;
+                    // At least one term in the search box must be in the option's title
+                    // or path. Could add check for if term is somewhere in category list too
+                    _.forEach(searchTerms, function (term) {
+                        var titleContains = title.indexOf(term) > -1,
+                            pathContains = searchablePath.indexOf(term) > -1;
+                        if (term !== "" && (titleContains || pathContains)) {
```

Comment:
Also, you probably want a newline between the var declarations and the if statement above.